### PR TITLE
Add guidance around code review process

### PR DESCRIPTION
**Pull request by jeanduplessis**
Signed-off-by: Jean du Plessis <jeandp@gmail.com>

### Description of your changes

Introduces guidance around the code review process requirements and expectations from PR authors and code reviewers. The intention is to clarify the process and expectations of how code reviews should work in the project.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

Not applicable.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: contributing/README.md
Comment (jeanduplessis) [inline contributing/README.md:164]: TODO source some good examples that showcase the guidelines in action

**Comment by jeanduplessis** [comment]
@ezgidemirel @phisco would you mind doing an initial review for me on this proposed change?

File: contributing/README.md
Comment (negz) [inline contributing/README.md:183]: I do think this is a great source of inspiration. I'm reluctant to adopt it though, as I find it a little rigid. I don't want readers to get the impression that our project _requires_ conventional comments.

File: contributing/README.md
Comment (negz) [inline contributing/README.md:178]: We could maybe remove this part. I think it's true, but I suspect most OSS projects are constantly in a "not everyone has internalized the contribution requirements" state due to the constantly shifting set of contributors.

**Comment by negz** [comment]
I've tagged @jbw976 to review this too - he's particularly good at giving thoughtful/kind review feedback.

File: contributing/README.md
Comment (jeanduplessis) [reply contributing/README.md:99]: I'm reframing this introduction paragraph to capture the spirit of using code reviews as an educational tool.

File: contributing/README.md
Comment (jeanduplessis) [reply contributing/README.md:183]: I'll remove for now.

File: contributing/README.md
Comment (jeanduplessis) [reply contributing/README.md:178]: Sounds good, removing.

**Review by ezgidemirel — APPROVED** [review]
Thanks for the PR @jeanduplessis! LGTM

File: contributing/README.md
Comment (negz) [inline contributing/README.md:195]: Is this effectively the same as the third entry?

File: contributing/README.md
Comment (negz) [inline contributing/README.md:214]: This reminds me that another tool available to us is to use a video call to review anything tricky or hard to communicate synchronously. Might be worth mentioning, but not a blocker.

**Review by negz — APPROVED** [review]
This is great, thanks @jeanduplessis! It's helpful to have this all written down.

File: contributing/README.md
Comment (jeanduplessis) [reply contributing/README.md]: You're right. I've combined the two into one.

File: contributing/README.md
Comment (jeanduplessis) [reply contributing/README.md:213]: I was thinking about adding this, and then more thoughts came up regarding do we mention using Slack calls or just allow folks to self-arrange (likely the best), also what is the threshold for requesting a sync review, and what if reviewees start asking for sync calls too often. In the end I decided to hold off from adding it and feel we can add it in future if we start seeing a pattern emerging that makes sense to encourage.

**Comment by jeanduplessis** [comment]
@jbw976 this one is ready for your review now.

**Review by phisco — APPROVED** [review]
LGTM!

File: contributing/README.md
Comment (jbw976) [inline contributing/README.md:131]: i got confused here by "reviewee" and read it as "reviewer" - what do you think about using terminology of "contributor"/"author" (the person opening the PR) and "reviewer" (the person reviewing the PR)? 🤔 

Looks like we are using "author" in the section header below, so maybe standardizing on that would be good?

File: contributing/README.md
Comment (jbw976) [inline contributing/README.md:148]: You can consider also including a specific incentive for reviewing PRs here: it's a nice path to becoming a project maintainer.

File: contributing/README.md
Comment (jbw976) [inline contributing/README.md:155]: typo: "by by" (repeated "by")

File: contributing/README.md
Comment (jbw976) [inline contributing/README.md:154]: I expected to see some emphasis here on opening quality PRs that thoroughly explain what they are trying to accomplish, how they were tested, etc.

However, I see that type of material is covered in details below in the [`#contributing-code`](https://github.com/crossplane/crossplane/blob/bd8798124ed91500f40e5b13f7ade38faab70500/contributing/README.md#contributing-code) section, so maybe it would be good to mention the concept briefly here and link to it for further reading?

File: contributing/README.md
Comment (jbw976) [inline contributing/README.md:175]: possibly in this list we can reinforce "Are the changes well documented and their intent explained sufficiently for current and future readers of your code to easily understand it?", or something to that effect :)

**Review by jbw976 — APPROVED** [review]
This is a great set of guidelines for having effective and kind code reviews @jeanduplessis, love to see this type of cultural investment for the project.

I've left a few suggestions, all of which you can assess and incorporate or not as you see fit.

There's also a small typo catch too 🤓

File: contributing/README.md
Comment (jeanduplessis) [reply contributing/README.md:131]: Yip, I agree with being consistent. I'm updating it to "author".

File: contributing/README.md
Comment (jeanduplessis) [reply contributing/README.md:148]: Adding the following sentence: "In addition, reviewing PRs is a path to becoming a maintainer on the project."

File: contributing/README.md
Comment (jeanduplessis) [reply contributing/README.md:155]: Ah good catch, thanks.

File: contributing/README.md
Comment (jeanduplessis) [reply contributing/README.md:154]: I'm bringing attention to this by adding the following: 

_While preparing your PR, be mindful of the instructions and requirements in the
[contributing code](#contributing-code) section._

File: contributing/README.md
Comment (jeanduplessis) [reply contributing/README.md:175]: Adding a slightly more brief version: "Are the changes well documented, and their intent explained sufficiently for current and future readers?"

**Comment by jeanduplessis** [comment]
Thanks for the review, @jbw976.

@crossplane/crossplane-maintainers this PR has received the required approvals and is now ready for merging.

**Review by turkenh — APPROVED** [review]
Thanks @jeanduplessis  🙌

**Pull request merged**
